### DISCUSSION TITLE
"Message body empty"

### DISCUSSION THREAD

**Discussion by pingo700** [PHPMailer/PHPMailer#2692] [Q&A]
Hello, I have a problem when sending an html via PHPmailer, the body is correct and is sent to some users. The following message appears in $mail->ErrorInfo "Message body empty".

**Comment by Synchro** [comment]
It will be your code, but you'll need to show it to us before we can help.

**Reply by pingo700** [reply]
$mail = new PHPMailer();
        $body = $this->Body;
        $mail->From     = $this->From;
        $mail->FromName = $this->FromName;
        $mail->AddAddress($this->Fromto, $this->NameAdd);
        $mail->isHTML(true);
        $mail->AddCC($this->From, $this->NameCC);
        $mail->IsHTML(true);
        $mail->CharSet  = 'utf-8';
        $mail->Subject  = $this->Subject;
        $mail->AltBody  = $this->Subject;
        $mail->MsgHTML($body);
        $mail->IsSMTP();
        $mail->isHTML(true);
        $mail->SMTPSecure = 'tls';
        $mail->SMTPAuth = true;
        $mail->Host     = $this->Host;
        $mail->Port     = $this->Port;
        $mail->Username = $this->Username;
        $mail->Password = $this->Password;
        if ($mail->Send()) {
            return 1;
        } else {
            return $mail->ErrorInfo;
        }

**Comment by Synchro** [comment]
From the syntax, it looks like you adapted this code from a *very* old example, so make sure you're using the latest version of PHPMailer.

This is a bit of mess. A large source of confusion is  that you're setting `Body` to `$this->Body`, but then later on calling `msgHTML($body)`, which also sets the `Body` property (as well as `AltBody`, which will overwrite the `Subject` that you put in there). So pick *one* of these mechanisms: either set `Body` and `AltBody` directly, or use `msgHTML()` – but don't do both. Also, you only need to call `isHTML()` once, not 3 times!

All that doesn't explain the symptom though because we can't see what's in `$this->Body` – if you can show that it contains a non-empty value at the same time as receiving the empty body error, we might have something to start from. Try doing this before your `send` to confirm what you've done:

    var_dump($this->Body, $mail->Body);
